File: escenario1/src/escenario1/Vehicle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package escenario1;

/**
 *
 * @author Dell
 */
abstract class Vehicle {
    
    protected int cylinderCapacity;
    protected int modelYear;
    protected double valuation;

    public Vehicle(int cylinderCapacity, int modelYear, double valuation) {
        this.cylinderCapacity = cylinderCapacity;
        this.modelYear = modelYear;
        this.valuation = valuation;
    }
            
    abstract double calculateRegistrationCost();
       
}

File: src/Car/Vehicle.ts
export abstract class Vehicle {
  protected cylinderCapacity: number;
  protected modelYear: number;
  protected valuation: number;

  public constructor(
    cylinderCapacity: number,
    modelYear: number,
    valuation: number,
  ) {
    this.cylinderCapacity = cylinderCapacity;
    this.modelYear = modelYear;
    this.valuation = valuation;
  }

  abstract calculateRegistrationCost(): number;
}
